src/main/scala/org/broadinstitute/dsde/workbench/sam/service/TosService.scala

```diff
@@ -19,19 +19,43 @@ class TosService (val directoryDao: DirectoryDAO, val registrationDao: Registrat
   val termsOfServiceFile = "termsOfService.md"
 
   def resetTermsOfServiceGroupsIfNeeded(): IO[Option[BasicWorkbenchGroup]] = {
+
     if(tosConfig.enabled) {
-      getTosGroup().flatMap {
-        case None =>
-          logger.info(s"creating new ToS group ${getGroupName()}")
-          val groupEmail = WorkbenchEmail(s"GROUP_${getGroupName(tosConfig.version)}@$appsDomain")
-          directoryDao.createGroup(BasicWorkbenchGroup(WorkbenchGroupName(getGroupName(tosConfig.version)), Set.empty, groupEmail), None, SamRequestContext(None)).flatMap { createdGroup =>
-            registrationDao.disableAllHumanIdentities(SamRequestContext(None)).map { _ => Option(createdGroup)}
-          }
-        case group => IO.pure(group)
+      //this is kinda gnarly, but i'd rather leave the if else totally separate to easy in later code-cleanup after this if branch is defunct
+      if(tosConfig.version == 1) {
+        getTosGroup().flatMap {
+          case None =>
+            logger.info(s"creating new ToS group ${getGroupName()}")
+            val groupEmail = WorkbenchEmail(s"GROUP_${getGroupName(tosConfig.version)}@$appsDomain")
+            directoryDao.createGroup(BasicWorkbenchGroup(WorkbenchGroupName(getGroupName(tosConfig.version)), Set.empty, groupEmail), None, SamRequestContext(None)).flatMap { createdGroup =>
+              if(tosConfig.enforced) {
+                directoryDao.loadGroup(WorkbenchGroupName(getGroupName(tosConfig.version)), SamRequestContext(None)).map {
+                  case Some(group) => group.members
+                  case None => Set[WorkbenchSubject]()
+                } flatMap { exemptedUsers =>
+                  registrationDao.disableAllHumanIdentities(SamRequestContext(None), exemptedUsers).map { _ => Option(createdGroup) }
+                }
+              } else IO.pure(Option(createdGroup))
+
+            }
+          case group => IO.pure(group)
+        }
+      } else {
```

Comment:
this `else` is essentially the previous implementation, with special logic to handle the `enforced` flag